src/algebraic_geometry/properties.lean

```diff
@@ -245,4 +597,326 @@ begin
     { intro x, exact x.rec _ } }
 end
 
+lemma is_integral_of_is_irreducible_is_reduced [is_reduced X] [H : irreducible_space X.carrier] :
+  is_integral X :=
+begin
+  split,
+  intros U hU,
+  split,
+  { intros a b e,
+    simp_rw [← basic_open_eq_bot_iff, ← opens.not_nonempty_iff_eq_bot],
+    by_contra h,
+    push_neg at h,
+    exfalso,
+    obtain ⟨_, ⟨x, hx₁, rfl⟩, ⟨x, hx₂, e'⟩⟩ := @@nonempty_preirreducible_inter _ H.1
+      (X.to_LocallyRingedSpace.to_RingedSpace.basic_open a).2
+      (X.to_LocallyRingedSpace.to_RingedSpace.basic_open b).2
+      h.1 h.2,
+    replace e' := subtype.eq e',
+    subst e',
+    replace e := congr_arg (X.presheaf.germ x) e,
+    rw [ring_hom.map_mul, ring_hom.map_zero] at e,
+    apply @zero_ne_one (X.presheaf.stalk x.1),
+    rw ← is_unit_zero_iff,
+    convert hx₁.mul hx₂,
+    exact e.symm },
+  exact (@@LocallyRingedSpace.component_nontrivial X.to_LocallyRingedSpace U hU).1,
+end
+
+lemma is_integral_of_open_immersion {X Y : Scheme} (f : X ⟶ Y) [H : is_open_immersion f]
+  [is_integral Y] [nonempty X.carrier] : is_integral X :=
+begin
+  constructor,
+  intros U hU,
+  have : U = (opens.map f.1.base).obj (H.base_open.is_open_map.functor.obj U),
+  { ext1, exact (set.preimage_image_eq _ H.base_open.inj).symm },
+  rw this,
+  haveI : is_domain (Y.presheaf.obj (op (H.base_open.is_open_map.functor.obj U))),
+  { apply_with is_integral.component_integral { instances := ff },
+    apply_instance,
+    refine ⟨⟨_, _, hU.some.prop, rfl⟩⟩ },
+  exact (as_iso $ f.1.c.app (op $ H.base_open.is_open_map.functor.obj U) :
+    Y.presheaf.obj _ ≅ _).symm.CommRing_iso_to_ring_equiv.is_domain _
+end
+
+
+universes u v
+
+/-- The function field of an integral scheme is the local ring at its generic point. -/
+noncomputable
+abbreviation Scheme.function_field [is_integral X] : CommRing :=
```

Comment:
I think it makes sense to define this for irreducible schemes. Of course the `field` instance will require integrality.